lib/mempool/mempool.js

```diff
@@ -513,8 +513,17 @@ class Mempool extends EventEmitter {
     if (!entry)
       return false;
 
-    if (this.isSpent(hash, index))
-      return false;
+    const spender = this.getSpent(hash, index);
+    // If the TX that spent this coin signals
+    // replaceability, then even though it is "spent"
+    // we still "have" it.
+    if (spender) {
```

Comment:
I think `hasCoin` becomes a not-ideal name for this if it depends on many things. Also this will ruin compatibility with the `getCoin` - even though it is not used for anything important.

One solution to this (and others I will mention) could be to pass new flag "rbf" to make it old behavior the default one and only use rbf when necessary and make it clearer. Alternatively, change the name of the method ?